### ISSUE TITLE
LibraryImportAttribute incorrectly stating use of extension method and preventing compilation

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/runtime#90211]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/LibraryImportAttribute-incorrectly-stati/10415101)._

---
I am using .NET 8 Preview 6 and Visual Studio 2022, updated today.

I updated a library today, changing from DllImport to LibraryImport, but it indicates CS0755.

Here is my imported method declaration:

```
public unsafe static partial class CppCoreFuncs
{
    #region private
    [LibraryImport(DLL, EntryPoint = "Poly2Cofs_multi_buf"), SuppressUnmanagedCodeSecurity]
    private static partial FuncResult Poly2CofsMultiBuf(void* buffer, InputMode mode);
    #endrgion

...
}
```

This generates the following code:

```
public unsafe static partial class CppCoreFuncs
{
    [System.Runtime.InteropServices.DllImportAttribute("CPP Core.dll", EntryPoint = "Poly2Cofs_multi_buf", ExactSpelling = true)]
    private static extern partial global::RWT. ROCT. Core.Public.FuncResult Poly2CofsMultiBuf(void* buffer, global::RWT. ROCT. Core.Public.InputMode mode);
}
```

When I compile, I get this:

_Error CS0755 Both partial method declarations must be extension methods or neither may be an extension method... Core\Microsoft.Interop.LibraryImportGenerator\Microsoft.Interop.LibraryImportGenerator\LibraryImports.g.cs_

There is no extension method. If I change back to DLLImport then the next method import declaration with LibraryImport gets this error. It only appears once.

---
#### Feedback Bot on 7/14/2023, 04:28 AM: 

(private comment, text removed)
#### Ian Campbell on 8/4/2023, 00:17 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by AaronRobinsonMSFT** [comment]
@jtschuster Any chance you could take a look at this?

**Comment by jtschuster** [comment]
I can't get a repro with the information provided. Here's the setup I've tried tinkering with:

Class.cs:

```C#
using System.Runtime.InteropServices;
using System.Runtime.CompilerServices;
using System.Security;
using RWT.ROCT.Core.Public;

[assembly: DisableRuntimeMarshallingAttribute]

public unsafe static partial class CppCoreFuncs
{
    const string DLL = "CPP Core.dll";
    #region private
    [LibraryImport(DLL, EntryPoint = "Poly2Cofs_multi_buf"), SuppressUnmanagedCodeSecurity]
    private static partial FuncResult Poly2CofsMultiBuf(void* buffer, InputMode mode);
    #endregion
}
namespace RWT.ROCT.Core.Public
{
    enum InputMode {}
    struct FuncResult {}
    // enum FuncResult {}
}
```

Repro.csproj:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
    <OutputType>Library</OutputType>
    <TargetFramework>net8.0</TargetFramework>
    <Nullable>enable</Nullable>
    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
  </PropertyGroup>

</Project>
```

global.json

```json
{
  "sdk": {
    "version": "8.0.100-preview.6.23330.14"
  }
}
```

**Comment by IamIC** [comment]
This issue is resolved in 17.8.0 Preview 1.0.
